core/app/models/spree/payment_method.rb

```diff
@@ -17,7 +17,10 @@ class PaymentMethod < Spree::Base
     scope :active, -> { where(active: true) }
     scope :available_to_users, -> { where(available_to_users: true) }
     scope :available_to_admin, -> { where(available_to_admin: true) }
-    scope :available_to_store, -> (store) { (store.present? && store.payment_methods.empty?) ? self : store.payment_methods }
+    scope :available_to_store, ->(store) do
+      raise ArgumentError, "You must provide a store" if store.nil?
+      store.payment_methods.empty? ? all : where(id: store.payment_method_ids)
+    end
```

Comment:
The whole method is not very clearly written. It's sort of all in-reverse. I think it should be just:

```ruby
scope :available_to_store, -> (store) { where(store: store) }
```

That should also work if store is `nil`.

The `raise ArgumentError` is not much improvement over the Undefined method `payment_methods` on `Nil:Class` that one would usually get.

I think the whole method